telemetry/definitions/commonDefinitions.json

```diff
@@ -1468,6 +1468,11 @@
             "type": "string",
             "description": "A generic version metadata"
         },
+        {
+            "name": "windowId",
```

Comment:
Could `source` be used for this intent? I assume not, because sometimes a metric would have an associated window _and_ an upstream reason for the event.

I think we should generalize this slightly, by renaming it to `viewId`. Some windows may host multiple views, and we should be able to attribute events to those views.

Reply:
> Could `source` be used for this intent?

`source` is more of a "semantic" concept (it describes a UI element or feature name) whereas (I assume) windowId/viewId will be an integer.

Reply:
Good points.

> Could `source` be used for this intent?

The reason for why you assumed we cannot is correct.

> I think we should generalize this slightly, by renaming it to `viewId`

Would it make sense to have both `windowId` and `viewId`?

`windowId` - the ID of the IDE window
`viewId` - the ID of the view within the `windowId`

Eg:
```json
{
  "windowId": "abc-123",
  "viewId": "amazon-q-chat"
}
```

A `viewId` will be static I would think, but the `windowId` will be dynamic.

---

As for using `source` I think this would not be reuseable for either as in VSC we use it for other purposes such as "which function caused this metric to emit", it is a generalize field.

Reply:
It makes sense to me that we can't just use `source`

I could see defining both window and view as metadata. I think this is going to be confusing for developers though ("which field do I use and what do they mean?"), and if we can boil it down to a singular field, we're more likely to avoid errors caused by ambiguity. Since a window _is_ a view, can we have a single field, but if a product wants to be specific, they can codify that in the field? 

For example, if the list of cloudwatch log streams was being shown within both the Lambda Function Window and the CloudWatch Log Groups Window, you'd differentiate using a viewId like "lambda.cwlstreams" and "cwlgroups.cwlstreams".

Reply:
Yeah makes sense to me. So in my case would this match your expectation for it to look something like:

`viewId: "38061227-ee4a-4be2-b869-5521fad9f0731724683347178.lambda.cwlstreams"`
vs
`viewId: "38061227-ee4a-4be2-b869-5521fad9f0731724683347178.cwlgroups.cwlstreams"`

Where the UUID is the IDE window ID, but then we append the underlying "view" id

Reply:
This examples match how I imagine the field would be used.

I think all of the toolkits differ in implementation details, but not all systems are aware of the view they are performing operations for. How are you planning to obtain these values without requiring coupling between view and controller?

Reply:
As for VSC, we can currently wrap a function in a "context" with our span/run mechanisms. This allows a high level caller such as the view to set a `viewId` field, and then deep within the call stack when we eventually need to resolve the `viewId` it will return the value set by the high level call. This is a cleaner alternative to just passing `viewId` as an argument to every nested function call.

Also I think for now, we would just set the window UUID in the `viewId` field, and then later look to solve to problem of also attaching the "view" source to it.

Reply:
little late to the party but we identify UI elements with `elementId` but it's not used until now as uuid. I feel it's fine if we are can reuse

Reply:
This should be at compute level right? if I'm VSC local user and I run multiple ec2 from local env, we should be able to identify these ec2s. how about `computeEnvId` / `computeId` or `instanceId`?

Reply:
why can't we use `instanceId` and `elementId` as separate fields, since elementId is designed to point a particular ui element. is it because you are adding this field to all metrics? (along with product, os, version?)

Reply:
The previous discussion regarding a view ID for a specific view within an IDE is not relevant to this PR anymore. See Chris' comments down below.

The plan is to add a `sessionId` adjacent to `clientId` which is a unique ID to a specific IDE window. So if there are other windows that are ssh'd to EC2 for example, they will have their own unique sessionId. On top of that they will have the `ec2` computeEnv so we can know how many different EC2 specific IDE windows that are using